src/storages/sled_storage/alter_table.rs

```diff
@@ -134,8 +131,14 @@ impl AlterTable for SledStorage {
                 tree.insert(temp_new_key, new_schema_key.as_bytes())?;
             }
 
-            Ok(())
-        })
+            Ok(TxPayload::Success)
+        });
+
+        match self.check_retry(tx_result) {
+            Ok(true) => self.rename_schema(table_name, new_table_name).await,
+            Ok(false) => Ok((self, ())),
+            Err(e) => Err((self, e)),
+        }
```

Comment:
Make decl macro..

```rust
macro_rules! check_retry {
    ($self:expr, $tx_result:expr, $method:ident ($($params:expr),*)) => {{
        match $self.check_retry($tx_result) {
            Ok(true) => $self.$method($($params),*).await,
            Ok(false) => Ok(($self, ())),
            Err(e) => Err(($self, e)),
        }
    }};
}
```

Why don't you write it this way?
The pattern seems to repeat itself.

`check_retry!(self, tx_result, rename_schema(table_name, new_table_name))`

Reply:
I agree that could be a good choice to use.
However, I'm actually trying to reduce using macros now because decl macros are convenient but it can also make hard to track code coverage.
If we use Rust closures or functions, Rust compiler can verify its behavior, but macro is more likely a simple tool for generating codes.
To verify codes, it is enough to verify each function or closure, but for the case of macros, we still need to check all the codes generated by macros.

Here in transaction implementation, it was especially tough due to a lot of concurrency issues.
I depend a lot on [tarpaulin](https://github.com/xd009642/tarpaulin), it provide code coverage test for Rust.
e.g.
![image](https://user-images.githubusercontent.com/2025065/130025568-e9ed9fe5-411f-4058-97a9-736a236dd910.png)

![image](https://user-images.githubusercontent.com/2025065/130025832-422e7491-6a10-416d-a2b0-19cc3e4c61fb.png)

It is very convenient, I can easily check which codes are covered or not.
(fyi. I usually run `cargo tarpaulin --out Html --output-dir {dir-to_export}` to generate HTML format report and review it).

However, if I use macro to pack these codes, then I'm still supposed to test each code branches but I cannot get proper support from `tarpaulin`.

It can be really cool if we can make `match -- check_retry` part as a shared function, but because of `$self.$method($($params),*).await,` this part... it's a bit tough.
I didn't know what to do so just left like now. but... maybe I can make the function.. by wrapping `$self.$method($($params),*).await,` as a closure and passing it to a shared function to handle it.
Sounds... quite interesting

Reply:
I agree that it is difficult to track code. 
I will make it a shared function and deliver it to you.

Reply:
You can also upload as a separate PR. This becomes really a good motivation to do something.
I think we can also apply similar approach to remove `try_`s, `try_self!`, `try_block!` and `try_into!`. Now.. I think I can replace it into normal functions.

Reply:
Okay, I'll put it in a separate PR